historical_system_profiles/baseline_service_interface.py

```diff
@@ -0,0 +1,30 @@
+from urllib.parse import urljoin
+
+from historical_system_profiles import metrics
+from kerlescan import config
+from kerlescan.service_interface import fetch_url, internal_auth_header
+from kerlescan.constants import AUTH_HEADER_NAME, INTERNAL_BASELINE_SVC_ENDPOINT
+
+
+def fetch_system_baseline_associations(system_id, service_auth_key, logger):
+    """
+    call baseline systems association endpoint to get a list of baselines
+    this system is associated with
+    """
+
+    auth_header = {**{AUTH_HEADER_NAME: service_auth_key}, **internal_auth_header()}
+
+    internal_baselines_location = urljoin(
+        config.baseline_svc_hostname, INTERNAL_BASELINE_SVC_ENDPOINT
+    )
+
+    query = "?system_id=%s" % system_id
+    url = internal_baselines_location % query
```

Comment:
I missed this previously... its similar to the other url building code.  I think the import thing is the "?" and "&" are part of the path passed into the urljoin in case they need to be escaped, etc.  Maybe there is a different urllib module/method that would also support the value substition as this would add better security around url building.

Reply:
This is solved with update of kerlescan